tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -185,15 +189,48 @@ async def with_ci_credentials(context: ConnectorTestContext, gsm_secret: Secret)
     return ci_credentials.with_env_variable("VERSION", "dev").with_secret_variable("GCP_GSM_CREDENTIALS", gsm_secret).with_workdir("/")
 
 
-async def with_ci_connector_ops(context: ConnectorTestContext) -> Container:
+async def with_ci_connector_ops(context: PipelineContext) -> Container:
     """Installs the ci_connector_ops package in a Container running Python > 3.10 with git..
 
     Args:
-        context (ConnectorTestContext): The current test context, providing the repository directory from which the ci_connector_sources sources will be pulled.
+        context (PipelineContext): The current test context, providing the repository directory from which the ci_connector_sources sources will be pulled.
 
     Returns:
         Container: A python environment container with ci_connector_ops installed.
     """
     python_base_environment: Container = with_python_base(context, "python:3-alpine")
     python_with_git = python_base_environment.with_exec(["apk", "add", "gcc", "libffi-dev", "musl-dev", "git"])
     return await with_installed_python_package(context, python_with_git, CI_CONNECTOR_OPS_SOURCE_PATH, exclude=["pipelines"])
+
+
+def with_poetry(context: PipelineContext) -> Container:
+    """Installs poetry in a python environment.
+
+    Args:
+        context (PipelineContext): The current test context, providing the repository directory from which the ci_credentials sources will be pulled.
+
+    Returns:
+        Container: A python environment with poetry installed.
+    """
+    python_base_environment: Container = context.dagger_client.container().from_("python:3-alpine")
+    python_with_git = python_base_environment.with_exec(["apk", "add", "gcc", "libffi-dev", "musl-dev", "git"])
+    python_with_poetry = python_with_git.with_exec(INSTALL_POETRY_PACKAGE_CMD)
+
+    poetry_cache: CacheVolume = context.dagger_client.cache_volume("poetry_cache")
+    poetry_with_cache = python_with_poetry.with_mounted_cache("/root/.cache/pypoetry", poetry_cache, sharing=CacheSharingMode.PRIVATE)
```

Comment:
@alafanechere I opted not to share cache for this, good idea or bad idea?

Reply:
I you want the cache to by reused throughout builds you should use LOCKED or SHARED.
If concurrent access is problematic you should chose LOCKED, if it's not problematic use SHARED as it will allow concurrent pipeline to access it without waiting.